main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -558,6 +580,27 @@ object Verifier {
     case _ => failMismatchedShape(rec, "Record", loc)
   }
 
+  /**
+    * Convert from the java class `klazz` to the corresponding `MonoType`
+    */
+  private def javaClassToMonoType(klazz: Class[_]): MonoType = klazz.getName match {
```

Comment:
I'm not fully sure it is a good solution, but could you try going the other way? We know the whole set of expected Monotypes, so match the expected type, and then assert what the class should be (by name, or something else)

Reply:
maybe call it `checkJavaEq`